backend/app/services/btc_detail.py

```
@@ -0,0 +1,204 @@
+
+    open_legs = derive_open_legs(
+        [position],
+        {ticker: current_price},
+        option_marks=marks,
+        profit_review_pct=rules.management.profit_review_pct,
+        dte_review_days=rules.management.dte_review_days,
+        expiration_warning_days=rules.management.expiration_warning_days,
+    )
+
+    leg = next(
+        (l for l in open_legs if str(l["id"]) == str(leg_id)), None
```

Ruff (0.15.13)

[error] Ambiguous variable name: `l`

(E741)